### DISCUSSION TITLE
UI for renaming tags

### DISCUSSION THREAD

**Discussion by tobias-klein** [ezra-bible-app/ezra-bible-app#447] [General]
The UI for renaming tags is currently optimized for mouse-based use (desktop). The [Rename] link/button shows up when doing mouseover in the tag list.

How can we properly support this usecase in a touch-based UI?

**Options:**
1) Permanently show the [Rename] link for each tag. (Would clutter the UI ...)
2) Show a context menu button instead of delete buttons. Move the rename and delete actions into a context menu.
3) Show an edit button for each tag. When clicking the button, fill the current box with a UI to edit the tag, but also provide a possibility to go back to the tag list.

Any other options?

@zhuiks @augustin-colesnic What do you think?

**Comment by augustin-colesnic** [comment]
@tobias-klein for me the most natural UX would be to click / tap directly on the tag itself to trigger edit mode  (modal window). Like in the attached jpg and get rid of the [rename] option on hover  (the red crossed in the example jpg attached)
![tag edit  mode UX](https://user-images.githubusercontent.com/592416/134044900-2d5ea403-7d62-41a3-9f53-c59e7f6794ea.jpg)

**Reply by tobias-klein** [reply]
You can use the "click/tap on tag title" as the method for assigning a tag (once you have selected at least one verse before). I think it's easier to hit the tag title than the checkbox to the left of it, especially on a touch screen.

**Comment by augustin-colesnic** [comment]
By the way: What is the purpose of the disabled checkbox next to delete button here? I fail to understand what is it used for.
<img width="1488" alt="Screenshot 2021-09-20 at 20 26 31" src="https://user-images.githubusercontent.com/592416/134046910-695bf211-87b1-4ae5-8eaf-f5ca2930ccaf.png">

**Reply by tobias-klein** [reply]
That checkbox is showing the assignment of a tag! Currently you can assign/unassign tags either by clicking on the checkbox or on the tag title. The checkbox is only enabled once you have previously clicked on a verse or selected multiple verses.

**Comment by tobias-klein** [comment]
I meanwhile think the third option would be best. A tag details editor component could later also be used to change the categories of a tag and to show some additional meta data.

@zhuiks Any additional feedback from you regarding this topic?

**Comment by zhuiks** [comment]
I think that the editing should be in place. Without popup dialog. It can be triggered either by single/double tap (I think it's pretty common) or by "pencil" icon or both.
Instead of the checkbox, we can use "tag" icon that would be highlighted if the tag is assigned to the currently selected verse.

![Знімок екрану з 2021-10-04 15-31-35](https://user-images.githubusercontent.com/3584119/135929076-7fa2f755-d5ef-44e5-be7a-d4a18aee1ccc.png)

![Знімок екрану з 2021-10-04 15-39-45](https://user-images.githubusercontent.com/3584119/135929083-38c83f66-dd68-42d2-9036-5becc536b094.png)
I also think that "garbage can" icon is better for tag deletion than the "x" icon. The "x" icon can be confused with tag unassigment action.

**Reply by tobias-klein** [reply]
I like this proposal! I just turned this into issue #475.
